google/cloud/aiplatform/metadata/metadata.py

```diff
@@ -52,20 +52,29 @@ def run_name(self) -> Optional[str]:
             return self._run.display_name
         return None
 
-    def set_experiment(self, experiment: str):
+    def set_experiment(self, experiment: str, description: Optional[str] = None):
         """Setup a experiment to current session.
 
         Args:
             experiment (str):
                 Required. Name of the experiment to assign current session with.
+            description (str):
+                Optional. Description of an experiment.
         Raises:
             ValueError if a context with the same name as the experiment is create but with a different schema.
+            Or if exp_description is set but experiment is not.
         """
 
+        if experiment is None:
+            raise ValueError(
+                "Experiment name needs to be set in order to add experiment descriptions."
```

Comment:
I think this exception may be more informative if it was moved up the call stack to init with remediation included ie. `Experiment needs to be set using aiplatform.init`

Reply:
I was originally thinking to support update description if experiment was previous set through init so that check logic would need to be in metadata.py, but later changed my mind on that..... Agree it is better to handle the exception in init with current logic. I also added a couple tests to cover the exception.